src/components/widget/SelectionDropdown.js

```diff
@@ -155,17 +155,17 @@ export default class SelectionDropdown extends Component {
           navigate(false);
           break;
         case 'Escape':
+          event.preventDefault();
           onCancel();
           break;
         case 'Enter':
+          event.preventDefault();
           onSelect(selected);
           break;
         default:
           return;
       }
     }
-
-    event.preventDefault();
```

Comment:
Could you explain how this fix works? At least from what I see the only thing changed is the order of calling `preventDefault`.

Should `preventDefault` also not be added after [here](https://github.com/metasfresh/metasfresh-webui-frontend/blob/27b2967090f29f964eb3e03e0735e6b01a6cb9f3/src/components/widget/SelectionDropdown.js#L146)?

Reply:
I was asking because I can't see anything that would depend on the order of when `preventDefault` is called, as long as it happens synchronously in the handler. Am I wrong here?

> Nope. We want tab to work, and we want char key presses to be bubbled up to parent components.

There is a `return` in the `default` case, so all key presses that are not handled do not get a `preventDefault` and bubble up

Reply:
Yup, and that's exactly what we want. Everything APART of those keys should be handled by parent components (if available) as we don't care about them here.

By the order I meant the how the events are handled by DOM.